os/adding-swap.md

```diff
@@ -0,0 +1,90 @@
+# Adding swap in CoreOS/
+Swap is the process of moving pages of memory to a designated part of the hard disk, freeing up space when needed. Swap can be used to alleviate problems with low-memory environments.
+
+By default CoreOS does not include a partition for swap, however one can configure their system to have swap.
+
+## Manually adding a swapfile on ext4
+
+Run these commands **as root** to create a 512 megabyte swapfile.
+
+```
+$ mkdir /var/vm/
+$ fallocate -l 1024m /var/vm/1GiB.swap
+$ chmod 600 /var/vm/1GiB.swap
+$ mkswap /var/vm/1GiB.swap
+$ swapon /var/vm/1GiB.swap
+```
+
+**Note**: The partition on which you are creating the swapfile must meet the following requirements:
+
+* Must not be of type **tmpfs**, **btrfs**, or **xfs**.
+* Must be large enough for your swapfile size.
+
+To verify that our swapfile meets those requirements, run the following command:
+
+```
+$ df -Th
+Filesystem     Type      Size  Used Avail Use% Mounted on
+[...]
+/dev/sdXN      ext4      2.0G  3.0M  1.8G   1% /var
+```
+
+The block device `/dev/sdXN` the correct filesystem type and has enough space for a 1GiB swapfile, so everything should work.
+
+### Create a unit to initialize the swap on boot
+
+The following systemd unit file handles creating, starting, and stopping the swapfile `/etc/systemd/system/swap.service`.
```

Comment:
The following systemd unit file creates, starts, and stops the swapfile. It should be written to `/etc/systemd/system/swap.service`.